common/models/types/index.d.ts

```diff
@@ -135,23 +135,6 @@ declare interface LexicalModel {
    */
   predict(transform: Transform, context: Context): Distribution<Suggestion>;
 
-  /**
-   * Tokenizes the current context state, returning an array of all resulting tokens.
-   * @param context 
-   */
-  tokenize(context: Context): USVString[];
-
-  /**
-   * Performs a wordbreak operation given the current context state, returning whatever word
-   * or word fragment exists that starts before the caret but after the most recent whitespace
-   * preceding the caret.  If no such text exists, the empty string is returned.
-   * 
-   * This function is designed for use in generating display text for 'keep' `Suggestions`
-   * and display text for reverting any previously-applied `Suggestions`.
-   * @param context 
-   */
-  wordbreak(context: Context): USVString;
```

Comment:
This is technically a breaking change wrt Keyman 13; however, I'm not reallly sure it matters? 🤔

Reply:
This is the exact reason behind the response I gave [above](https://github.com/keymanapp/keyman/pull/3706#issuecomment-709713784).

> I tested it with previously-compiled MTNT, no changes required. It's mostly a change to the model templates (the TrieModel class used by MTNT) and their connections to the LMLayer, not specific compiled models.
>
> So, the internal connections will have changed from 12.0 to 14.0, but only internal code not specific to models. It probably would affect any custom models... if they existed.

This is the part that would break custom models.

Reply:
tl;dr: Is there a way to keep the interface working for custom models that may use this? It's okay to add a compiler deprecation warning but it's not okay to break the interface.

There may be no published custom models in the repository, but that doesn't mean they don't exist -- there are many, many keyboards not in our repository, and there will be models that are not in our repository also.

Generally, I won't approve breaking changes in interfaces for models and keyboards: a 13.0 model MUST continue to work with 14.0, 15.0 and into the future. Keyman 6.0 keyboards from the 90's work with Keyman 14.0. In theory, Keyman 4 keyboards from 1996 would work even though I don't know if any still exist (and wouldn't be Unicode so would definitely be 'legacy').

One reason for this is that keyboards and models are distributed to users, and users may not be able to acquire a new version of the package from the package developer, nor downgrade to an earlier version of Keyman to continue using their package. An upgrade of Keyman must not break existing packages.

We cannot assume an evergreen software world for Keyman, where everything is always updated.

This does not necessarily mean that the model MUST compile unmodified with 14.0 compiler, although that is certainly a hugely important goal, and we should be very careful not to make unnecessary interface changes. It does mean that a model compiled with 12.0 MUST run unmodified with 13.0 or 14.0 LMLayer. I would prefer to see a deprecation warning in the compiler for a few versions before an interface disappears.

Finally, any change like this must also be documented in the [release notes](https://github.com/keymanapp/keyman/wiki/Keyman-13.0-to-14.0-Migration).